topkapi.go

```diff
@@ -64,25 +65,35 @@ func NewTopK(k, approxCorpusSize uint64, delta float64) (*Sketch, error) {
 }
 
 func newSketch(b, l uint64) *Sketch {
+	return &Sketch{
+		l:      l,
+		b:      b,
+		sparse: make(map[string]uint64, b),
```

Comment:
can we have the sparse map size explicit?
even an explicit constructor maybe
NewTopKWithSparseMap(...)
that way the current behaviour would remain the same
and the map would only be used when you use this new constructor

Reply:
we have `l` rows each with `b` buckets
we are keeping up to `b` words in sparse mode

Reply:
I am more for having an "automatic good choice" then having the caller know about intrinsics of the algorithm. That said, it's not crystal clear to me why b is the right number. I can see why it must scale with b - but why not b/2 or 4*b?

Reply:
you are right again! it should be a fraction of b... but a fraction of b would not gain us anything
we want to guarantee for the same size 100% accuracy so ideally we have a map that grows to l*b words, because after that we are probabilistic right?

Reply:
If n is the number of inserted words then topkapi is probabilistic from n > 1 since hash collisions are just handled with evictions :-) From n > b we are 100% sure to have hash collisions since we are inserting each word in every row.
I am thinking maybe something like int(0.8*b) ...?